tests/unit_tests/epee_utils.cpp

```diff
@@ -1757,79 +1757,103 @@ TEST(parsing, number)
   // the parser expects another character to end the number, and accepts things
   // that aren't numbers, as it's meant as a pre-filter for strto* functions,
   // so we just check that numbers get accepted, but don't test non numbers
+  // We set is_float/signed_val to the opposite of what we expect the result to
+  // make sure that match_number2 is changing the bools as expected
 
   s = "0 ";
   i = s.begin();
+  is_float_val = true;
```

Comment:
I think it would be more readable and less error-prone to split match_number and match_number2 into separate tests and structure the `number2` cases as a loop over a sequence of "test scenario" structs consisting of the input and the expected out values. This way, you clarify how to exercise the code under test (preconditions and postconditions), have clear rules for state initialization, and it's trivial to add tests as warranted. The test assertions can include the input value as the message in order to make the failure clear, rather than using line numbers.

I realize that perhaps you ended up with copy/paste because that was the original design, but at this point the refactor creates *less* noise because the original code was much briefer and somewhat incorrect, so applying the corrections in a single location makes the review simpler, as opposed to copy/pasting many lines into many places.

Reply:
@iamamyth I'm deleting `match_number()`, as it is unused, so this should be the only unit test for `parsing.number`. Looping through a list of structs is good idea though, I think I'll do that